src/test/mir-opt/generator-tiny/rustc.main-{{closure}}.generator_resume.0.mir

```diff
@@ -1,5 +1,5 @@
 // MIR for `main::{{closure}}#0` 0 generator_resume
-// generator_layout = GeneratorLayout { field_tys: [HasDrop], variant_fields: [[], [], [], [_0]], storage_conflicts: BitMatrix { num_rows: 1, num_columns: 1, words: [1], marker: PhantomData } }
```

Comment:
oh this is a nice side effect from running the deaggregator before `StateTransform`

Reply:
What is the effect? Less stuff in the generator?
Cc @jonas-schievink

Reply:
Looks like it the deaggregator "unpeels" empty types now, so this ZST doesn't need to be stored anymore (not that it matters too much for ZSTs).

Note that the generator transform uses the liveness dataflow analysis, which doesn't work correctly after the deaggregator has run - it treats every assignment to a field of an aggregate as a use of the base local, which may lead to locals being considered live up to the function's entry point. This does not cause unsoundness here, but it might cause some size regressions. However it looks like the `MaybeRequiresStorage` analysis fully mitigates this effect by only requiring storage for a local after the first field of the aggregate is written.